note: ‘snprintf’ output 9 bytes into a destination of size 8

I noticed this warning while building the code natively:
```
searchresultsdialog.cpp: In member function ‘QString SearchResultsDialog::downloadName() const’:
searchresultsdialog.cpp:67:26: warning: ‘snprintf’ output truncated before the last format character [-Wformat-truncation=]
         snprintf(tmp, 8, "%8.8x", i++);
                          ^~~~~~~
searchresultsdialog.cpp:67:17: note: ‘snprintf’ output 9 bytes into a destination of size 8
         snprintf(tmp, 8, "%8.8x", i++);
         ~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~
```

Seem like something to address.

--- a/searchresultsdialog.cpp
+++ b/searchresultsdialog.cpp
@@ -1,6 +1,7 @@
 #include <stdio.h>
 
 #include <QtGui>
+#include <QByteArray>
 #include <QStringListModel>
 #include <QDir>
 #include <QFile>
@@ -63,9 +64,7 @@ QString SearchResultsDialog::downloadName() const
     unsigned i = 0;
     QString fileName;
     do {
-        char tmp[9];
-        snprintf(tmp, 8, "%8.8x", i++);
-        tmp[8] = '\0';
+        QByteArray tmp = QByteArray::number(i++, 16).rightJustified(8, '0');
         fileName = QDir(dir).absoluteFilePath(QString(tmp) + ".epub");
     } while (QFile(fileName).exists());
     qDebug() << fileName;

